frontend/projects/upgrade/src/app/features/dashboard/feature-flags/modals/upsert-feature-flag-modal/upsert-feature-flag-modal.component.ts

```diff
@@ -152,7 +155,14 @@ export class UpsertFeatureFlagModalComponent {
 
   // Close the modal once the feature flag list length changes, as that indicates actual success
   listenForFeatureFlagGetUpdated(): void {
-    this.subscriptions.add(this.isSelectedFeatureFlagUpdated$.subscribe(() => this.closeModal()));
+    this.subscriptions.add(
+      this.isSelectedFeatureFlagUpdated$.subscribe(() => {
+        this.closeModal();
+        if (this.isContextChanged) {
+          this.router.navigate(['/featureflags']);
```

Comment:
Why should we navigate to featureflags page?

Reply:
Can't we make the modal wait until the deletion is done before it is closed? I think the page should remain on the details page for best user experience. There's no reason to navigate to the root page after editing a feature flag on the details page.